contracts/infrastructure/salary/PayrollClerk.sol

```diff
@@ -56,32 +56,33 @@ contract PayrollClerk is PayrollClerkStorage {
     return workers.length;
   }
 
+  function getPrice(address token) internal view returns (uint256) {
+    return IPriceCalculator(calculator()).getPriceWithDefaultOutput(token);
+  }
+
   function multiplePay(address[] calldata _workers, uint256[] calldata _workedHours)
   external onlyControllerOrGovernance {
     require(_workers.length == _workedHours.length, "wrong arrays");
-    for (uint256 i = 0; i < _workers.length; i++) {
-      pay(_workers[i], _workedHours[i]);
+    for (uint256 h = 0; h < tokens.length; h++) {
+      uint256 tPrice = getPrice(tokens[h]);
+      for (uint256 i = 0; i < _workers.length; i++) {
+        require(baseHourlyRates[_workers[i]] != 0, "worker not registered");
+        pay(_workers[i], _workedHours[i], tokens[h], tPrice);
+      }
     }
   }
 
-  function pay(address worker, uint256 _workedHours) public onlyControllerOrGovernance {
-    require(baseHourlyRates[worker] != 0, "worker not registered");
-
-    uint256 totalSalaryUsd;
-    for (uint256 i = 0; i < tokens.length; i++) {
-      (uint256 salaryUsd, uint256 salaryToken) = computeSalary(worker, _workedHours, tokens[i]);
-      require(salaryToken <= ERC20(tokens[i]).balanceOf(address(this)), "not enough fund");
-      IERC20(tokens[i]).safeTransfer(worker, salaryToken);
-      totalSalaryUsd = totalSalaryUsd.add(salaryUsd);
-    }
+  function pay(address worker, uint256 _workedHours, address token, uint256 tPrice) internal onlyControllerOrGovernance {
```

Comment:
you can remove onlyControllerOrGovernance from internal function

Reply:
Removed! I also changed `computeSalary` to `internal`.